src/EditorFeatures/Core/GoToDefinition/GoToDefinitionHelpers.cs

```diff
@@ -101,6 +101,13 @@ internal static class GoToDefinitionHelpers
             bool thirdPartyNavigationAllowed = true,
             bool throwOnHiddenDefinition = false)
         {
+            var languageServices = project.Solution.Workspace.Services.GetLanguageServices(LanguageNames.FSharp);
+            var symbolicNavigationService = languageServices?.GetService<ISymbolicNavigationService>();
+            var found = symbolicNavigationService?.TryNavigateToSymbol(symbol, cancellationToken)
+                .WaitAndGetResult(cancellationToken);
```

Comment:
note: looking at this now, i realize that i would have a deep perf concern here.  Effectively, we're allowing *anyone* to participate in goto-def, and they might be super expensive.  For example, given a metadata-name, can F# even answer quickly if it can take over?

Reply:
To answer your question whether or not F# can answer this quickly:

Currently I'm matching the symbol's assembly name with the projects in the solution. If it _does not_ match an F# project I return immediately. If it does match an F# project I try to find the symbol (this is combarable to 'find all references' in the project) then return whether we found it (or not). If we did find it, we take control and navigate to the symbol.

Note that looking for the symbol currently is comparable to the work of 'find all references' but that in the future this could possibly improve. Since we have the namespace of the symbol here, we could keep a cache of <namespace, files> and only look in the files that match our namespace.

What do you think?